usb/esp_tinyusb/tinyusb_ext_phy.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-FileCopyrightText: 2020-2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "tinyusb_ext_phy.h"
+
+static usb_phy_handle_t phy_hdl;
+
+esp_err_t tinyusb_ext_phy_new(const tinyusb_config_t *config)
+{
+    // Configure USB PHY
+    usb_phy_config_t phy_conf = {0};
+
+    phy_conf.controller = USB_PHY_CTRL_OTG;
+    phy_conf.otg_mode = USB_OTG_MODE_DEVICE;
+
+    if (config->external_phy) {
+        // External PHY IOs config
+        usb_phy_ext_io_conf_t ext_io_conf = { 0 };
+        ext_io_conf.vp_io_num = USBPHY_VP_NUM;
+        ext_io_conf.vm_io_num = USBPHY_VM_NUM;
+        ext_io_conf.rcv_io_num = USBPHY_RCV_NUM;
+        ext_io_conf.oen_io_num = USBPHY_OEN_NUM;
+        ext_io_conf.vpo_io_num = USBPHY_VPO_NUM;
+        ext_io_conf.vmo_io_num = USBPHY_VMO_NUM;
```

Comment:
I've already deprecated these GPIO numbers (see description of MR !28293 for more details).

Basically, with an external FSLS PHY, those signals can be routed to any GPIO via the GPIO matrix. Thus the current use of the `USBPHY_..._NUM` macros are forcing users to use an arbitrary set of GPIOs, whereas the cleaner way to do it is to have the users provide the list of GPIOs they want to use via a configuration structure.